### DISCUSSION TITLE
JAX suddenly 2x slower on V100 GPUs

### DISCUSSION THREAD

**Discussion by minqi** [jax-ml/jax#14061] [General] [answered]
Seemingly overnight, my JAX project (in v. 0.4.1) became suddenly ~2x slower in terms operations per second when running on V100 and Quadro GP100s. The project git commit hash is identical and there were no changes to my conda environment. Does anyone have any ideas for what might cause this performance degradation?

**Comment by astanziola** [comment]
I am experiencing also increasing slowness of JAX since a few releases, expecially when working with very large array (on the order of a Gb or more)

**Comment by mattjj** [comment]
Hrm, this is weird...

> The project git commit hash is identical and there were no changes to my conda environment.

@minqi do you think your `jaxlib` changed? Could you try installing an older vs current `jaxlib`?

Any chance at sharing a repro?

**Comment by yashk2810** [comment]
Also, what change did you make when you noticed this slowness? Was it a jax/jaxlib upgrade or was it just shifting from 1 GPU to another GPU or something else?

If you upgraded JAX, don't forget to bump your jaxlib to 0.4.1 too!

**Comment by minqi** [comment, marked as answer]
Thanks for the responses! I ended up reinstalling both jax + jaxlib and now see the prior training speeds.

**Reply by mattjj** [reply]
That's great! Perhaps that's the first line of defense: if you come across this discussion page, update your jax and jaxlib first!
